src/components/GlobalLeaderboards/Context.tsx

```diff
@@ -0,0 +1,213 @@
+  const [positionsHash, setPositionsHash] = useState<string>();
+  const [accountsHash, setAccountHash] = useState<string>();
+  const topAccountsHeaderClick = useCallback((key: keyof TopAccountsRow) => () => {
+    if (key === "wins") {
+      setAccountsOrderBy(accountsOrderBy === "wins" ? "losses" : "wins");
+      setAccountsOrderDirection(1);
+    } else if (key === accountsOrderBy) {
+      setAccountsOrderDirection((d: number) => -1 * d);
+    } else {
+      setAccountsOrderBy(key);
+      setAccountsOrderDirection(1);
+    }
+  }, [accountsOrderBy]);
+
+  const topPositionsHeaderClick = useCallback((key: keyof TopPositionsRow) => () => {
+    if (key === positionsOrderBy) {
+      setPositionsOrderDirection((d: number) => -1 * d);
+    } else {
+      setPositionsOrderBy(key);
+      setPositionsOrderDirection(1);
+    }
+  }, [positionsOrderBy]);
+
+  const nextPositionsHash = utils.sha256(topPositionsUnordered.data ? topPositionsUnordered.data.map(a => (
+    a[positionsOrderBy]!.toString().split("").map((c: string) => c.charCodeAt(0))
+  )).flat() : []);
+
+  if (positionsHash !== nextPositionsHash) {
+    setPositionsHash(nextPositionsHash);
+  }
+
+  useEffect(() => {
+    if (!topPositionsUnordered.data) {
+      return;
+    }
+
+    const data = [...topPositionsUnordered.data].sort((a, b) => {
+      const key = positionsOrderBy;
+      if (a[key] instanceof BigNumber && b[key] instanceof BigNumber) {
+        return positionsOrderDirection * ((a[key] as BigNumber).gt(b[key] as BigNumber) ? -1 : 1);
+      } else {
+        // TODO: remove log
+        // eslint-disable-next-line no-console
+        console.warn("Not a BigNumber", { key, a, b });
+        return 1;
+      }
+    }).map((p, i) => ({ ...p, rank: i }));
+
+    setTopPositions(data);
+    if (p.current && !p.current.pSorted) {
+      p.current("sorted top positions");
+      p.current.pSorted = true;
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [positionsHash, positionsOrderBy, positionsOrderDirection]);
+
+  const nextAccountsHash = utils.sha256(topAccountsUnordered.data ? topAccountsUnordered.data.map(a => (
+    a[accountsOrderBy]!.toString().split("").map((c: string) => c.charCodeAt(0))
+  )).flat() : []);
+
+  if (accountsHash !== nextAccountsHash) {
+    setAccountHash(nextAccountsHash);
+  }
+
+  useEffect(() => {
+    if (!topAccountsUnordered.data) {
+      return;
+    }
+
+    const data = [...topAccountsUnordered.data];
+    data.sort((a, b) => {
+      const key = accountsOrderBy;
+      if (a[key] instanceof BigNumber && b[key] instanceof BigNumber) {
+        return accountsOrderDirection * ((a[key] as BigNumber).gt(b[key] as BigNumber) ? -1 : 1);
+      } else {
+        // TODO: remove log
+        // eslint-disable-next-line no-console
+        console.warn("Not a BigNumber", {key, a, b});
+        return 1;
+      }
+    });
+
+    data.forEach((a, i) => a.rank = i);
```

Comment:
looks stylish but a bit not clear that you modify some field here, actually entries in topAccountsUnordered will be modified which comes from other hook, if i'm not mistaken. 

i think .map would be safer